lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -375,7 +375,7 @@ namespace Js
 
     ScriptContext::~ScriptContext()
     {
-        Assert(isFinalized || !isInitialized);
+        Assert(isFinalized);
```

Comment:
shouldn't revert. We can get rid of the whole assert. 
I was afraid we might miss some scenario where we would delete scriptContext is wrong time. I think we cover all three now and the assert can be removed:
. finalize time when the scriptcontext is held by library
. half way during initialization of scriptcontext when we OOM; in both browser & JSRT scenario
. JSRT only, fail half way during JSRTContext initialization, but after scriptcontext is fully created.